reactor-core/src/test/java/reactor/core/publisher/MonoReduceSeedTest.java

```diff
@@ -155,6 +158,28 @@ public void accumulatorReturnsNull() {
 		  .assertError(NullPointerException.class);
 	}
 
+	@Test
+	public void onNextAndCancelRace() {
+		List<Integer> list = new ArrayList<>();
+		final AssertSubscriber<Object> testSubscriber = AssertSubscriber.create();
+
+		MonoReduceSeed.ReduceSeedSubscriber<Integer, List<Integer>> sub =
+				new MonoReduceSeed.ReduceSeedSubscriber<>(testSubscriber,
+						(l, next) -> {
+							l.add(next);
+							return l;
+						}, list);
+
+		sub.onSubscribe(Operators.emptySubscription());
+
+		//the race alone _could_ previously produce a NPE
+		RaceTestUtils.race(() -> sub.onNext(1), sub::cancel);
```

Comment:
Is this helper basically syncing up two threads to execute the lambdas? The situation with race tests is that they may not happen just running once, therefore, I usually loop the whole test 1.000-10.000 times to be way more certain.

Reply:
that's what I did initially, but then since I'm manipulating a subscriber directly I figured I could simply trigger the same conditions as in the race by doing another `sub.onNext` afterwards. I could have removed the RaceTestUtils usage entirely I guess...